[sushant-wayal-stringhash] New port sushant-wayal-stringhash

sushant-wayal opened the pull request:
# PR Description

This PR adds a new port: **`stringhash`**, a lightweight header-only C++ library for efficient and flexible string hashing.

`stringhash` provides a clean and simple API to compute rolling and polynomial hashes with customizable mod values, powers, and multiple hash layers.  
It’s designed for performance-critical use cases like substring comparison, pattern matching, and competitive programming.

## Highlights
- Header-only (`stringhash.h`)
- Supports both `int` and `long long` hashing
- Configurable number of hashes and base/mod pairs
- Easy integration via `#include <stringhash>`
- MIT licensed

---

## Checklist (new port)
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] Port name follows upstream convention (`stringhash`).
- [x] All optional dependencies resolved or disabled.
- [x] Versioning and license match upstream (`MIT`).
- [x] Copyright file installed.
- [x] Source verified from official GitHub release.
- [x] Usage text tested and accurate.
- [x] Version database updated with `./vcpkg x-add-version --all`.
- [x] Single version entry added.

---

## Notes
- Added `stringhash` v1.0.0 as an initial release.
- Verified build via `vcpkg install stringhash` on both Windows and Linux using CMake.

sushant-wayal committed:
Add stringhash to vcpkg

sushant-wayal committed:
Add stringhash to vcpkg

sushant-wayal committed:
version database

JavierMatosD reviewed (dismissed):
This PR still looks like it's in progress. I'm going to place this PR in draft while the build failures get addressed.

sushant-wayal committed:
updated stringhash to vcpkg

sushant-wayal committed:
version database

sushant-wayal committed:
Update stringhash portfile and versioning information

sushant-wayal committed:
Remove deprecated version 1.0.0 from stringhash.json

BillyONeal committed:
Fix line endings.

BillyONeal committed:
Revert accidential s2n change

sushant-wayal committed:
Rename port to sushant-wayal-stringhash and update to v1.1.0

- Renamed port from 'stringhash' to 'sushant-wayal-stringhash' to avoid conflicts with USD
- Updated to version 1.1.0 with new SHA512 hash
- Added proper CMake dependencies (vcpkg-cmake, vcpkg-cmake-config)
- Updated CMake config path to match new port name
- Cleaned versions database to contain only v1.1.0 entry
- Removed UWP restriction to support all platforms

sushant-wayal committed:
Resolve merge conflicts - complete port rename to sushant-wayal-stringhash

sushant-wayal committed:
Fix build failures by switching to header-only installation

- Removed CMake build configuration that was causing test failures
- Simplified to header-only file installation (no build required)
- Removed CMake dependencies (vcpkg-cmake, vcpkg-cmake-config)
- Headers are already properly structured in repository with sushant-wayal namespace
- This avoids CMake configuration mismatches and version conflicts

dg0yt commented:
The error after cmake installation was from wrong options to `vcpkg_cmake_config_fixup`:
~~~diff
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/sushant-wayal-stringhash)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/stringhash PACKAGE_NAME stringhash)
~~~

For reference, install log had:
~~~
-- Installing: /mnt/vcpkg-ci/p/sushant-wayal-stringhash_x64-linux/lib/cmake/stringhash/stringhashTargets.cmake
-- Installing: /mnt/vcpkg-ci/p/sushant-wayal-stringhash_x64-linux/lib/cmake/stringhash/stringhashConfig.cmake
-- Installing: /mnt/vcpkg-ci/p/sushant-wayal-stringhash_x64-linux/lib/cmake/stringhash/stringhashConfigVersion.cmake
~~~

BillyONeal committed:
LF-ize

BillyONeal reviewed (dismissed):
Thanks!

BillyONeal commented:
/azp run
